packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2302,34 +2308,37 @@ abstract class RenderBox extends RenderObject {
 
   @override
   void markNeedsLayout() {
-    if ((_cachedBaselines != null && _cachedBaselines!.isNotEmpty) ||
-        (_cachedIntrinsicDimensions != null && _cachedIntrinsicDimensions!.isNotEmpty) ||
-        (_cachedDryLayoutSizes != null && _cachedDryLayoutSizes!.isNotEmpty)) {
+    if ((_cachedBaselines?.isNotEmpty ?? false) ||
+        (_cachedIntrinsicDimensions?.isNotEmpty ?? false) ||
```

Comment:
fwiw and absolutely feel free to ignore this: the older version seems easier to read to me. ("if x is not null and x is not empty" as opposed to "if x is not empty unless it's null in which case treat that as null but replace null with false" or something)